protocol/contracts/templegold/SpiceAuction.sol

```diff
@@ -0,0 +1,252 @@
+    uint256 private immutable _deployTimestamp;
+
+    /// @notice Keep track of the different configurations for each auction
+    mapping(uint256 auctionId => SpiceAuctionConfig config) public auctionConfigs;
+    /// @notice Keep track of claimed amounts per auction token. This helps calculate amount of auction tokens for next epoch.
+    mapping(address token => uint256 amount) private _claimedAuctionTokens;
+    /// @notice Keep track of total allocation per auction token
+    mapping(address token => uint256 amount) private _totalAuctionTokenAllocation;
+
+    constructor(
+        address _templeGold,
+        address _spiceToken,
+        address _daoExecutor,
+        string memory _name
+    ) {
+        spiceToken = _spiceToken;
+        daoExecutor = _daoExecutor;
+        templeGold = _templeGold;
+        name = _name;
+        _deployTimestamp = block.timestamp;
+    }
+
+    /**
+     * @notice Set config for an epoch. This enables dynamic and multiple auctions especially for vested scenarios
+     * @dev Must be set before epoch auction starts
+     * @param _config Config to set
+     */
+    function setAuctionConfig(SpiceAuctionConfig calldata _config) external onlyDAOExecutor {
+        /// @dev epoch Id is only updated when auction starts. 
+        /// @dev cannot set config for past or ongoing auction
+        uint256 currentEpochIdCache = _currentEpochId;
+        if (currentEpochIdCache > 0) {
+            EpochInfo memory info = epochs[currentEpochIdCache];
+            /// Cannot set config for ongoing auction
+            if (info.startTime <= block.timestamp && block.timestamp < info.endTime) { revert InvalidConfigOperation(); }
+        }
+        if (_config.duration < MINIMUM_AUCTION_PERIOD) { revert CommonEventsAndErrors.InvalidParam(); }
+        // startCooldown can be zero
+        if (_config.waitPeriod == 0
+            || _config.minimumDistributedAuctionToken == 0) { revert CommonEventsAndErrors.ExpectedNonZero(); }
+        if (_config.recipient == address(0)) { revert CommonEventsAndErrors.InvalidAddress(); }
+
+        currentEpochIdCache += 1;
+        auctionConfigs[currentEpochIdCache] = _config;
+        emit AuctionConfigSet(currentEpochIdCache, _config);
+    }
+
+    /// @notice Remove auction config set for last epoch
+    function removeAuctionConfig() external override onlyDAOExecutor {
+        /// only delete latest epoch if auction is not started
+        uint256 epochId = _currentEpochId;
+        // Cannot reset an ongoing auction
+        EpochInfo storage info = epochs[epochId];
+        if (info.startTime <= block.timestamp && block.timestamp < info.endTime) { revert InvalidConfigOperation(); }
+        // use `epochId+1` for config of auction not started
+        SpiceAuctionConfig storage config = auctionConfigs[epochId+1];
+        if (config.duration == 0) { revert InvalidConfigOperation(); }
+        delete auctionConfigs[epochId];
```

Comment:
Let us image:
Epoch1 completed so `_currentEpochId` is 1 now.
And DAOExecutor set new AuctionConfig.
And if DAOExecutor calls `removeAuctionConfig` then it will delete the `Epoch1`, then Epoch1 bidders can not claim as no Epoch1 info.

Is this right logic?

Reply:
Good catch. 
There will still be `epochInfo` because that was not deleted, `auctionConfigs[epochId]` was deleted.
And yes that wil prevent the right claim if the wrong auction token is returned from `_getAuctionTokenForEpochStorage` if `config.isTempleGoldAuctionToken` gives `0` instead of `1` . 

Which will end up setting the claimed value for the wrong auction token with `_claimedAuctionTokens[auctionToken]`.

So user may claim wrong token